parsec/core/gui/desktop.py

```diff
@@ -40,3 +44,49 @@ def get_locale_language():
 def copy_to_clipboard(text):
     QGuiApplication.clipboard().setText(text, QClipboard.Clipboard)
     QGuiApplication.clipboard().setText(text, QClipboard.Selection)
+
+
+native_library = None
+
+
+def is_caps_lock_on():
+    # Apparently, there's no cross-platform way to check if caps lock is on in 2021.
+    # Qt doesn't provide any, neither does Python.
+    # We load native libraries and call native functions, GetKeyState on Windows
+    # and XkbGetIndicatorState on X11 systems.
+
+    global native_library
+
+    if platform.system() == "Windows":
+        try:
+            if not native_library:
+                native_library = cdll.LoadLibrary("User32")
+                native_library.GetKeyState.restype = c_short
+
+            VK_CAPITAL = c_int(0x14)
+            state = native_library.GetKeyState(VK_CAPITAL)
+            return state != 0
+        except:
+            return False
+    else:
+        try:
+            if not native_library:
+                if platform.system() == "Darwin":
+                    native_library = cdll.LoadLibrary("libX11.dylib")
+                else:
+                    native_library = cdll.LoadLibrary("libX11.so")
+                if not native_library:
+                    return False
+                native_library.XOpenDisplay.restype = c_void_p
+
+            state = c_uint()
+            XkbUseCoreKdb = c_uint(0x0100)
+            CapsLockMask = 0x1
+            display = native_library.XOpenDisplay(None)
+            if display:
+                native_library.XkbGetIndicatorState(display, XkbUseCoreKdb, byref(state))
```

Comment:
This line segfault on my machine (running from a conda environment):
```[1]    29154 segmentation fault (core dumped) ```

The library looks to be properly loaded:
```
<CDLL 'libX11.so', handle 55d874e894c0 at 0x7fd068505410>
```

I'll investigate it. Did you try it on another linux env ?

Reply:
Only on my VM running debian.